notes/2024/2024-10.md

```diff
@@ -0,0 +1,38 @@
+# Composite Schemas WG - October 2024
+
+**Watch the replay:**
+[Composite Schemas Working Group Meetings on YouTube](https://www.youtube.com/playlist?list=PLP1igyLx8foFjxyTg6wPn4pUkZwuAk2GR)
+
+## Oct 3 (Primary)
+
+### Independent notes (@smyrick)
```

Comment:
```suggestion
### Shared notes

* Summary of previous meetings
* Michael: GraphQLConf - skipped a couple meetings. Last week we focussed on
    * the source schema function
    * validation rules we need
    * the @key directive needs more specification text and examples
    * non-resolveable keys (hopefully we don't need these in the new spec, but we should write up about why we don't need it but Federation does)
    * batching chapter for GraphQL-over-HTTP (roughly 60% remaining)
    * enums: should everything be an intersection? Ongoing discussion
    * Overall we're focussed on validation
    * Next we'll focus on satisfiability (which is even more complex validation
*  Action items: none
* Shane: GraphQL Summit by Apollo
    * Speaking there: title is “Entities 201”
    * Want to highlight at the end of the talk on some of the WG progress
    * [Shane shows slides] high-level bullet points, who/what
    *  Three main things:
        * @lookup (no need for _entities)
        * batching (GraphQL-over-HTTP: request/variable batching)
        * standardized testing (the-guild-org/graphql-federation-gateway-audit; not official but we do want a standardized test suite and this looks like a great start)
    * Also mentioning a few of the other topics like @require @is @internal; but direct to WG to find out more.
* Michael: this is really good; the big thing that changes how we build the subgraphs is really @lookup.
* Shane: I'm going to avoid doing a deep dive technical thing. 
* [General agreement it should be a good talk]
* Pascal: enum merging RFC
    * Concern: when enums are shared across subgraphs - issues with input vs output.
    * RFC: describes the problem statement (errors when unrecognized, errors not surfaced early enough, challenging to evolve schema) with examples.
    * Solution 1: strict equality with @inaccessible flagged values to hide from supergraph.
        * Relies on server to not return inaccessible flagged values.
        * Easier evolvability.
    * Solution 2: context-based merging - strict equality if used in both input and output.
        * Better evolution if you're only using enum as input or only as output.
        * Complexity when enums are used in both directions
    * Solution 3: intersection merging
        * Only include enum values present in all subgraphs.
        * Non-obvious why certain values should not be returned (because they're not in the other subgraphs).
* Martijn: great write-ups and examples.
* Pascal: but the example has an intersection that is nothing, no values are shared.
* Pascal: composition errors occur when the intersection is empty.
* Michael: we'd love feedback on this - let us know which solution you prefer (or are there other solutions?)
* Derek: difference between 1 and 2 is that 1 requires explicit @inaccessible in subgraph-specific enum values.
* Benjie: …
* Michael: Hard to put default value into a public schema. Once you have a default value, is it gone/not needed?  How to think about collisions/naming?
* Shane: run into more issues: who decides the default value? Subgraph A/B?
* Pascal: encode as string in subgraph, then switch case in the destination graph?  If you want to add to an enum, you need a switch anyhow
* Shane: it always comes back to evolving schema on one subgraph, w/o being blocked by other subgraphs - primary use case everyone wants to solve - which is why we chose to go the route we have. Can we import external schema definitions?  Could be a platform feature
* Stephen: wonder if it’s worth a non-normative discussion w/in the spec around some of these aspects - workflow about how types get shared/rolled out - even some aspects of deploying/registering
    * There could be a "common types repo" which each of the subgraphs import from - enums would only be defined in the "common types repo".
    * The Graph should always remain in a good state regarding composition, so we shouldn't require atomic merges of subgraphs.
* Pascal: this would require two deployment cycles of all subgraphs - first with @inaccessible and then secondly removing it.
* Benjie (chat):
    * scalar MyEnum @specifiedBy('urn:graphql:scalar:enumhack:VAL_1,VAL_2,VAL_3')
    * ^ This would be on the subgraph, each one could have their own; supergraph would compose them all together, and each graph treats MyEnum as a String knowing that it might be VAL_1, VAL_2, VAL_3, or *any other string value*.
* Pascal: Schema governance was a topic at GraphQLConf - separate enums from input/output - if you enforce this at the org level, it Just Works (this is the Apollo way)

### Independent notes (@smyrick)
```